tests/Type/DefinitionTest.php

```diff
@@ -1585,84 +1597,6 @@ public function testDoesNotAllowIsDeprecatedWithoutDeprecationReasonOnEnum() : v
         $enumType->assertValid();
     }
 
-    /**
-     * Type System: List must accept only types
-     */
-    public function testListMustAcceptOnlyTypes() : void
-    {
-        $types = [
-            Type::string(),
-            $this->scalarType,
-            $this->objectType,
-            $this->unionType,
-            $this->interfaceType,
-            $this->enumType,
-            $this->inputObjectType,
-            Type::listOf(Type::string()),
-            Type::nonNull(Type::string()),
-        ];
-
-        $badTypes = [[], new stdClass(), '', null];
-
-        foreach ($types as $type) {
-            try {
-                Type::listOf($type);
-            } catch (Throwable $e) {
-                self::fail('List is expected to accept type: ' . get_class($type) . ', but got error: ' . $e->getMessage());
-            }
-        }
-        foreach ($badTypes as $badType) {
-            $typeStr = Utils::printSafe($badType);
-            try {
-                Type::listOf($badType);
-                self::fail(sprintf('List should not accept %s', $typeStr));
-            } catch (InvariantViolation $e) {
-                self::assertEquals(sprintf('Expected %s to be a GraphQL type.', $typeStr), $e->getMessage());
-            }
-        }
-    }
-
-    /**
-     * Type System: NonNull must only accept non-nullable types
-     */
-    public function testNonNullMustOnlyAcceptNonNullableTypes() : void
-    {
-        $nullableTypes    = [
-            Type::string(),
-            $this->scalarType,
-            $this->objectType,
-            $this->unionType,
-            $this->interfaceType,
-            $this->enumType,
-            $this->inputObjectType,
-            Type::listOf(Type::string()),
-            Type::listOf(Type::nonNull(Type::string())),
-        ];
-        $notNullableTypes = [
-            Type::nonNull(Type::string()),
-            [],
-            new stdClass(),
-            '',
-            null,
-        ];
-        foreach ($nullableTypes as $type) {
-            try {
-                Type::nonNull($type);
-            } catch (Throwable $e) {
-                self::fail('NonNull is expected to accept type: ' . get_class($type) . ', but got error: ' . $e->getMessage());
-            }
-        }
-        foreach ($notNullableTypes as $badType) {
-            $typeStr = Utils::printSafe($badType);
-            try {
-                Type::nonNull($badType);
-                self::fail(sprintf('Nulls should not accept %s', $typeStr));
-            } catch (InvariantViolation $e) {
-                self::assertEquals(sprintf('Expected %s to be a GraphQL nullable type.', $typeStr), $e->getMessage());
-            }
-        }
-    }
-
```

Comment:
Same for all of these tests, they might be still useful. Or am I missing something?

Reply:
`NonNull` constructor now has a `NullableType` type hint in its constructor, so PHP simply won't allow you to get it wrong (unless you configure your error reporting to ignore such errors). I've removed the asserts from the constructor, so these explicit exceptions are not thrown, and there is nothing to catch.

So, at this point, any tests along these lines would not be testing any internal logic in the framework--they would be testing PHP's type hinting, which seems like a waste of resources to me.

Reply:
OK, seems minor anyway